homeassistant/components/fritz/common.py

```diff
@@ -371,6 +378,29 @@ def scan_devices(self, now: datetime | None = None) -> None:
                 wan_access=None,
             )
 
+        if (
+            "Hosts1" not in self.connection.services
+            or "X_AVM-DE_GetMeshListPath"
+            not in self.connection.services["Hosts1"].actions
+        ) or (self._options and self._options.get(CONF_OLD_DISCOVERY, False)):
+            _LOGGER.debug(
+                "Using old hosts discovery method. (Mesh not supported or user option)"
+            )
+            self.mesh_role = MeshRoles.NONE
+            for mac, info in hosts.items():
+                if self.manage_device_info(info, mac, consider_home):
+                    new_device = True
+            self.signal(new_device)
+            return
+
+        try:
+            if not (topology := self.fritz_hosts.get_mesh_topology()):
+                raise Exception("Mesh supported but empty topology reported")
```

Comment:
since we are in a `try` block, which only catches `FritzActionError`, this raise will not be fetched so code will be further executed, but `topology` is `None` or just empty, this will led issues in L406 (`for node in topology.get("nodes", []):`)